### ISSUE TITLE
More minor fixes

### ISSUE THREAD

**Issue by sfayer** [ic-hep/pdm#259]
```
Traceback (most recent call last):
  File "/home/centos/pdm/src/pdm/utils/db.py", line 23, in managed_session
    yield request.db.session
  File "/home/centos/pdm/src/pdm/workqueue/WorkqueueDB.py", line 236, in update
    session.merge(self)
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/scoping.py", line 150, in do
    return getattr(self.registry(), name)(*args, **kwargs)
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/session.py", line 1604, in merge
    self._autoflush()
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/session.py", line 1208, in _autoflush
    util.raise_from_cause(e)
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/util/compat.py", line 199, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb)
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/session.py", line 1198, in _autoflush
    self.flush()
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/session.py", line 1919, in flush
    self._flush(objects)
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/session.py", line 2037, in _flush
    transaction.rollback(_capture_exception=True)
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/util/langhelpers.py", line 60, in __exit__
    compat.reraise(exc_type, exc_value, exc_tb)
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/session.py", line 2001, in _flush
    flush_context.execute()
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/unitofwork.py", line 372, in execute
    rec.execute(self)
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/unitofwork.py", line 526, in execute
    uow
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/persistence.py", line 65, in save_obj
    mapper, table, insert)
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/orm/persistence.py", line 570, in _emit_insert_statements
    execute(statement, multiparams)
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/engine/base.py", line 729, in execute
    return meth(self, multiparams, params)
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/sql/elements.py", line 322, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/engine/base.py", line 826, in _execute_clauseelement
    compiled_sql, distilled_params
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/engine/base.py", line 958, in _execute_context
    context)
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/engine/base.py", line 1159, in _handle_dbapi_exception
    exc_info
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/util/compat.py", line 199, in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb)
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/engine/base.py", line 928, in _execute_context
    context)
  File "/usr/lib64/python2.7/site-packages/sqlalchemy/engine/default.py", line 433, in do_executemany
    cursor.executemany(statement, parameters)
IntegrityError: (raised as a result of Query-invoked autoflush; consider using a session.no_autoflush block if this flush is occurring prematurely) (IntegrityError) columns id, job_id are not unique u'INSERT INTO jobelements (id, job_id, src_filepath, dst_filepath, listing, size, max_tries, attempts, timestamp, type, status) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)' ((1, 136, u'/~/bah/foo/test.out', None, None, 7216, 2, 0, '2018-06-01 15:16:19.703355', <JobType.REMOVE: 2>, <JobStatus.NEW: 0>), (1, 136, u'/~/bah/foo/', None, None, 22, 2, 0, '2018-06-01 15:16:19.703362', <JobType.REMOVE: 2>, <JobStatus.NEW: 0>))
```

**Comment by alexanderrichards** [comment]
> columns id, job_id are not unique

@sfayer this seems likely caused by the previous crashes as the DB looks to be in an unusual situation.

> (1, 136,

These are the job and element id values....136 elements. I don't think you had copied a directory at this stage. seems like multiple attempt have been populating the db and maybe confused it?

**Comment by alexanderrichards** [comment]
This is a stupid typo. I'll fix it as soon as I can, possibly over the weekend

**Comment by alexanderrichards** [comment]
closed by #262  and #263
